src/client/article.ts

```diff
@@ -110,6 +110,92 @@ function insertEpic(): void {
     }
 }
 
+interface FormData {
+    [key: string]: string;
+}
+
+function submit(body: FormData, form: Element): void {
+    fetch('https://callouts.code.dev-guardianapis.com/formstack-campaign/submit', {
+        method: 'POST',
+        body: JSON.stringify(body)
+    })
+    .then(() => {
+        form.innerHTML = "<p>Thank you for your contribution</p>"
+    })
+    .catch(() => {
+        const errorPlaceholder = form.querySelector('.error-message');
+        if (errorPlaceholder) {
+            errorPlaceholder.innerHTML = "Sorry, there was a problem submitting your form. Please try again later."
+        }
+    })
+}
+
+function readFile(file: Blob): Promise<string> {
+    return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        setTimeout(reject, 30000);
+
+        reader.addEventListener('load', () => {
+                if (reader.result) {
+                    const fileAsBase64 = reader.result
+                        .toString()
+                        .split(';base64,')[1];
+                    resolve(fileAsBase64);
+                }
+            }
+        );
+
+        reader.addEventListener('error', () => {
+            reject();
+        });
+
+        reader.readAsDataURL(file);
+    })
+}
+
+function callouts(): void {
+    const callouts = Array.from(document.querySelectorAll('details.callout'));
+    callouts.forEach(callout => {
+        const buttons = Array.from(callout.querySelectorAll('.callout-expand'));
+        buttons.forEach(button => {
+            button.addEventListener('click', () => {
+                callout.toggleAttribute('open');
+            })
+        })
+
+        const form = callout.querySelector('form');
+        if (!form) return;
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        form.addEventListener('submit', async (e): Promise<void> => {
+            try {
+                e.preventDefault();
+                const elements = form.elements as HTMLCollectionOf<HTMLInputElement>;
+                const data = Array.from(elements).reduce(async (o: Promise<FormData>, elem) => {
+                    const acc = await o;
+                    const { type, checked, name, value, files } = elem;
+                    if (type === 'radio') {
+                        if (checked) {
+                            acc[name] = value;
+                        }
+                    } else if (type === 'file' && files?.length) {
+                        acc[name] = await readFile(files[0]);
+                    } else if (value) {
+                        acc[name] = value;
+                    }
+                    return Promise.resolve(acc);
+                }, Promise.resolve({}));
+
+                submit(await data, form);
```

Comment:
I think this can be simplified slightly (and immutability retained up to a point):

```suggestion
                const data = Array.from(elements).reduce<Promise<FormData>>(async (formData, elem) => {
                    const { type, checked, name, value, files } = elem;

                    if (type === 'radio' && checked) {
                        return { ...(await formData), [name]: value };
                    } else if (type === 'file' && files?.length) {
                        return { ...(await formData), [name]: await readFile(files[0]) };
                    } else if (value) {
                        return { ...(await formData), [name]: value };
                    }
                    
                    return formData;
                }, Promise.resolve({}));
```

**Note** I haven't tested this, but it does type-check 🙂